### PR TITLE
bug/265 - enable working hour function and issue/272 - new UI for setting page to work with working hour function

### PR DESCRIPTION
**Pull request by brianzhang310**
- [ ] The pull request is complete according to the following criteria:
  - [ ] Acceptance criteria have been met
  - [ ] The documentation is kept up-to-date
  - [ ] Comprehensive tests (if applicable) have been generated and all pass.
  - [ ] The pull request describes the changes that have been made, and enough information is present in the description for any developer to understand what has changed
  - [ ] Commits have been squashed (or will be on merge).
  - [ ] The branch name is descriptive and follows the pull request title format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
  - [ ] The pull request title is of the following format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
  - [ ] The description uses [github syntax](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) to link to the issue. E,g Resolves se701g2/Doto#{Number}
  - [ ] At least two reviewers assigned. One of which must be the assigner of the issue.
  - [ ] If there are merge conflicts, run git rebase as opposed to git merge with master.

### CODE REVIEW

**Comment by brianzhang310** [comment]
A little summary 
1. bug/265, added a TaskShifter class which re-writes the start-time and end-time of tasks to make sure every task is managed in working time
2. issue/272, made some changes on setting page. now, you can see there is a save button to save the changes in working hours. If users don't save changes or quit the setting page without saving it, the working hour won't be saved.

File: doto-frontend/src/components/pages/Calendar/TaskShifter.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                    tasks[j].startDate.getTime() + (startActingHour - taskStart) * MILLISECONDS_PER_MINUTE,
+                );
+                tasks[j].endDate = new Date(
+                    tasks[j].endDate.getTime() + (startActingHour - taskStart) * MILLISECONDS_PER_MINUTE,
+                );
+            }
+        } else if (taskEnd > endActingHour) {
+            for (let j = i; j < tasks.length; j++) {
+                tasks[j].startDate = new Date(
+                    tasks[j].startDate.getTime() + (startActingHour + 1440 - taskStart) * MILLISECONDS_PER_MINUTE,
```
Comment (DeshmukhChinmay) [inline doto-frontend/src/components/pages/Calendar/TaskShifter.js:38]: Are you able to convert the magic numbers into a constant?

File: doto-frontend/src/components/pages/Settings/SettingsPage.js [hunk 2]
```diff
@@ -100,6 +121,32 @@ const WorkingHoursPicker = props => {
...
+                <Dialog
+                    open={dialog}
+                    onClose={handleClose}
+                    aria-labelledby="alert-dialog-title"
+                    aria-describedby="alert-dialog-description"
+                >
+                    <DialogTitle id="alert-dialog-title">{"Want to save those changes?"}</DialogTitle>
+                    <DialogContent>
+                        <DialogContentText id="alert-dialog-description">
+                            Your time table will be re-managed automatically. Please check again.
```
Comment (TCHE614) [inline doto-frontend/src/components/pages/Settings/SettingsPage.js:137]: The spelling of time table should be timetable or time-table. I also think the agree or disagree should just be Ok and Cancel instead. Other than that it seems fine to me

Comment (brianzhang310) [reply doto-frontend/src/components/pages/Settings/SettingsPage.js]: solved the grammar issue.

File: doto-frontend/src/components/pages/Calendar/TaskShifter.js (hunk 1 above)
Comment (brianzhang310) [reply doto-frontend/src/components/pages/Calendar/TaskShifter.js]: has refactored (startActingHour + 1440 - taskStart) to const shiftToTomorrow, and changed the colour of the save button.

File: doto-frontend/src/components/pages/Settings/SettingsPage.js (hunk 2 above)
Comment (TCHE614) [reply doto-frontend/src/components/pages/Settings/SettingsPage.js]: LGTM good job!

**Comment by brianzhang310** [comment]
@all-contributors please add @brianzhang310 for code, review, doc

**Pull request merged**